Convert `fanSpeed` and `saved_fan_speed` to `uint8_t`

gudnimg opened the pull request:
The fan speed variables only range from 0 to 255 and should be `uint8_t` (1 byte). Even in the planner/block this applies and sync the code a little bit with Marlin 2.

Change in memory (Multilang MK3S build):
Flash: -324 bytes
SRAM: -20 bytes

gudnimg committed:
Convert fanSpeed and saved_fan_speed to uint8_t

These variables only range from 0 to 255

For the menus we currently need to convert fanSpeed to int16_t.

Change in memory:
Flash: -160 bytes
SRAM: -2 bytes

gudnimg committed:
Change block buffer fan speed to one byte

It's only assigned to variables which are uint8_t

Change in memory:
Flash: -10 bytes
SRAM: -16 bytes

gudnimg commented:
Ran a MMU print with these changes. Print went very smooth and finished on time. Also tested editing the fan speed while printing, works fine as expected.

Note I also tweaked the PrusaSlicer profile a little bit to shrink the Wipe Tower.

![image](https://user-images.githubusercontent.com/8218499/221438779-534a766b-c1d5-4047-8022-e56faae57054.jpeg)

leptun commented on Firmware/ultralcd.cpp:
```diff
@@ -5581,7 +5581,7 @@ static void lcd_tune_menu()
 	MENU_ITEM_EDIT_int3_P(_T(MSG_BED), &target_temperature_bed, 0, BED_MAXTEMP - 10);
 
-	MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), &fanSpeed, 0, 255);//5
+	MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), (int16_t*)&fanSpeed, 0, 255);//5
```
This needs to be checked carefully. I am suspicious of that int16_t* cast. We need to check that the edit menu doesn't overwrite something around `fanSpeed`.

vintagepc commented on Firmware/ultralcd.cpp:
```diff
@@ -5581,7 +5581,7 @@ static void lcd_tune_menu()
 
-	MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), &fanSpeed, 0, 255);//5
+	MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), (int16_t*)&fanSpeed, 0, 255);//5
 	MENU_ITEM_EDIT_int3_P(_i("Flow"), &extrudemultiply, 10, 999);//6////MSG_FLOW c=15
```
Yes, this is definitely dangerous. You'll zero the adjacent byte in memory.

vintagepc commented on Firmware/ultralcd.cpp:
```diff
@@ -5732,7 +5732,7 @@ static void lcd_control_temperature_menu()
   MENU_ITEM_EDIT_int3_P(_T(MSG_BED), &target_temperature_bed, 0, BED_MAXTEMP - 3);
 #endif
-  MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), &fanSpeed, 0, 255);
+  MENU_ITEM_EDIT_int3_P(_T(MSG_FAN_SPEED), (int16_t*)&fanSpeed, 0, 255);
```
ditto

gudnimg committed:
Remove menu code in favor of simplicity

This change needs to be approved of course due to it affecting UI

The affect UI is only one menu
lcd_settings_linearity_correction_menu

I doubt many users are using this menu except very advanced users.

I propose we delete the variant of menu_draw_P which was used for
uint8_t but renders them as floats.
Instead treat
uint8_t same as int16_t i.e. just render integers.

Keeping it simple :) Solve our fanSpeed problem.

Change in memory:
Flash: -140 bytes
SRAM: -2 bytes

gudnimg commented:
If we can get that menu change [34c1e55](https://github.com/prusa3d/Prusa-Firmware/pull/3962/commits/34c1e55863650c3859567a2d73283109eefeb9d6) through that would eliminate one annoying edge case in the menu code. I didn't see before that it only affects one menu item.

Gives me more hope of finishing https://github.com/prusa3d/Prusa-Firmware/pull/3985 😆

gudnimg commented:
Rebased to sync with MK3 and fix conflict

DRracer commented on Firmware/menu.cpp:
```diff
@@ -452,20 +449,8 @@ void menu_draw_P<int16_t*>(char chr, const char* str, int16_t val)
-    }
-}
+template void menu_draw_P<int16_t>(char chr, const char* str, int16_t val);
+template void menu_draw_P<uint8_t>(char chr, const char* str, uint8_t val);
```
removal of the special code for linearity correction has been approved :tada:
